### PR TITLE
CORTX-29819: Consider REPAIRED state of devices while deciding pver health for bytecount

### PR DESCRIPTION
**Pull request by AbhishekSahaSeagate**
While traversing the conf tree of a pool version, don't count M0_NC_REPAIRED state
towards failed object count in the srecd vector. This behavior is only present when
m0_conf_pver_status() API is called and the rest of the motr is not affected.

Signed-off-by: Abhishek Saha <abhishek.saha@seagate.com>

# Problem Statement
In case of a drive failure, the pool version containing the failure was marked as degraded for bytecount, but upon SNS repair the pool version was not moved to healthy, so the bytecounts were still marked as degraded.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by AbhishekSahaSeagate** [comment]
@madhavemuri @nkommuri  can you please review this patch as per your convenience.

File: conf/pvers.c [hunk 1]
```diff
@@ -810,6 +810,31 @@ static int conf_pver_virtual_create(const struct m0_fid *fid,
 	return rc;
 }
 
+/**
+ * Builds a vector consisting of failed/offline objects in the conf tree of a
+ * pool version. Repaired devices are not considered to be failed.
+ */
+static int conf_pver_failed_recd_build(struct m0_conf_obj *obj, void *args)
```
Comment (nkommuri) [inline conf/pvers.c:817]: How about renaming function name "conf_pver_failed_recd_build" to "conf_pver_nonhealty_recd_build"?

File: conf/pvers.c [hunk 2]
```diff
@@ -810,6 +810,31 @@ static int conf_pver_virtual_create(const struct m0_fid *fid,
...
+{
+	uint32_t            *recd = args;
+	struct m0_conf_objv *objv;
+	unsigned             lvl;
+
+	if (m0_conf_obj_type(obj) == &M0_CONF_OBJV_TYPE) {
+		objv = M0_CONF_CAST(obj, m0_conf_objv);
+		if (!M0_IN(objv->cv_real->co_ha_state,
+			  (M0_NC_ONLINE, M0_NC_REPAIRED))) {
```
[the remark this answers is not in the document]
Comment (AbhishekSahaSeagate) [reply conf/pvers.c]: Done.

File: conf/pvers.c (hunk 1 above)
Comment (AbhishekSahaSeagate) [reply conf/pvers.c]: Done.

**Comment by AbhishekSahaSeagate** [comment]
@nkommuri @mehjoshi , can you please review this PR as per your convenience.

**Review by mehjoshi — APPROVED** [review]
LGTM.

**Comment by nkommuri** [comment]
@madhavemuri  Is following state sequence possible for a device, in case of direct rebalance?
Online --> Failed->Rebalancing->Online

If so, Rebalancing doesn't necessarily mean that repair is done and the bytes are healthy.
@AbhishekSahaSeagate apologies for the confusion.

**Pull request merged**